### ISSUE TITLE
problems when installing oidc-agent on macOS Monterey

### ISSUE THREAD

**Issue by gminiell** [indigo-dc/oidc-agent#428]
Hi, I quite new with this kind of installations on mac and I experienced a weird thing after installing oidc agent following instructions here https://github.com/indigo-dc/oidc-agent
Firstly I got this when submitting oidc-agent command:

miniello@Giorgias-MacBook-Pro ~ % oidc-agent
OIDC_SOCK=/var/folders/ty/4hdgyx4x1fg_d9q3ysf50p_h0000gn/T/oidc-npIH3x/oidc-agent.46434; export OIDC_SOCK;
OIDCD_PID=46459; export OIDCD_PID;
echo Agent pid $OIDCD_PID

and when I tried to start service I got this :

miniello@Giorgias-MacBook-Pro ~ % oidc-agent-service start
/opt/homebrew/bin/oidc-agent-service: line 77: /usr/bin/oidc-agent: No such file or directory
OIDC_SOCK=/var/folders/ty/4hdgyx4x1fg_d9q3ysf50p_h0000gn/T//oidc-agent-service/501/oidc-agent.sock; export OIDC_SOCK;
OIDCD_PID=; export OIDCD_PID;
OIDCD_PID_FILE=/var/folders/ty/4hdgyx4x1fg_d9q3ysf50p_h0000gn/T//oidc-agent-service/501/oidc-agent.pid; export OIDCD_PID_FILE;
/bin/cat /var/folders/ty/4hdgyx4x1fg_d9q3ysf50p_h0000gn/T//oidc-agent-service/501/oidc-agent.pid

so basically I'm not able to start the service. 
Could you help with this?
Thank you in advance
Giorgia

**Comment by shallmann** [comment]
(I am also just a user, but) running `eval $(oidc-agent)` should do

**Comment by gminiell** [comment]
Thank u very much for the quick reply. I did not have the chance to test it earlier, but it works!!! :)

**Comment by zachmann** [comment]
Hi @gminiell 
sorry for the delayed response - I was on vacation.
Thanks @shallmann for the help.

Indeed the agent can be started as noted with using `eval`.
Background: The other tools need to locate the agent's socket to communicate with it. The location of this socket is stored in an environment variable. Setting the environment variable cannot be done from the agent. Instead it prints the command needed to do so on startup. Through `eval` they are automatically executed.

The `oidc-agent-service` bash script eases some aspects, especially usage across different terminals and restarting it. In Linux this is integrated with Xsession, so that the agent it automatically available in all terminals in the session.
On MacOS we don't have such an integration. However, I don't see a reason why `oidc-agent-service` shouldn't work. (In principle)

In your case it did not work because the script did not find `/usr/bin/oidc-agent` because homebrew installs oidc-agent to a different location.
I have a fix for this that will be released in the next release.

Then it should be possible to to have a good integration by putting something like
```
eval `oidc-agent-service use`
```
Into your `.profile` or probably better `.bashrc`. 
Note that I'm not too familiar with bash and terminals with MacOS so I might get something wrong.

**Comment by gminiell** [comment]
Thank you very much @zachmann
